src/Nacos.cs

```diff
@@ -0,0 +1,64 @@
+﻿using Ocelot.Values;
+using Service = Ocelot.Values.Service;
+
+namespace Ocelot.Discovery.Nacos;
+
+public class Nacos : IServiceDiscoveryProvider
+{
+    private readonly INacosNamingService _client;
+    private readonly string _serviceName;
+
+    public Nacos(string serviceName,INacosNamingService client)
+    {
+        _client = client;
+        _serviceName = serviceName;
+    }
+
+    public async Task<List<Service>> GetAsync()
+    {
+        try
+        {
+            var instances = await _client.GetAllInstances(_serviceName)
+                .ConfigureAwait(false);
+
+            return instances?
+                .Where(i => i.Healthy && i.Enabled && i.Weight > 0) // 健康检查过滤
+                .Select(TransformInstance)
+                .ToList() ?? new List<Service>();
+        }
+        catch (NacosException ex)
+        {
+            throw ex;
+        }
+    }
+
+    private Service TransformInstance(Instance instance)
+    {
+        var metadata = instance.Metadata ?? new Dictionary<string, string>();
+
+        return new Service(
+            id: instance.InstanceId,
+            hostAndPort: new ServiceHostAndPort(instance.Ip, instance.Port),
```

Comment:
```suggestion
            hostAndPort: new(instance.Ip, instance.Port),
```
What is the scheme? Is the scheme known at this stage of building service?

Reply:
When a client fetches a service, it will access the corresponding service based on the address information at the time of registration, so if the registration is for an HTTPS address, then the fetched service is HTTPS, and vice versa.

Reply:
So, in the Nacos service discovery, scheme (protocol) management is not required, right?
For example, in Kubernetes we can manage the protocol while registering a service instance in the K8s registry.
In the next round of development, we will write acceptance tests that will show real usage in practice.